### PR TITLE
Feature/install output

### PR DESCRIPTION
**Pull request by lasote**
- [x] Refer to the issue that supports this Pull Request.  Partially address: #1864 #1895 #1998 
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/docs

- Missing docs. 
- Probably missing tests and slow review, @memsharded you could take charge of it if you want/can.
- I think it could be mature enough to be included as an experimental feature in 1.2 but feedback is welcome

### CODE REVIEW

File: conans/client/proxy.py [hunk 1]
```diff
@@ -96,6 +99,8 @@ def get_package(self, package_ref, short_paths):
             output.success('Already installed!')
             installed = True
             log_package_got_from_local_cache(package_ref)
+            if self._recorder:
```
Comment (memsharded) [inline conans/client/proxy.py]: this idiom all over the place ``if self._recorder`` seems like could be improved

File: conans/client/proxy.py [hunk 2]
```diff
@@ -390,10 +413,24 @@ def _retrieve_remote_package(self, package_ref, package_folder, output, remote=N
...
+                self._recorder.package_install_error(package_ref, INSTALL_ERROR_MISSING, msg,
+                                                     remote.url)
             return False
+        except RequestException as e:
+            if self._recorder:
+                self._recorder.package_install_error(package_ref, INSTALL_ERROR_NETWORK, str(e),
+                                                     remote.url)
+            raise e
+        except Exception as e:
```
Comment (memsharded) [inline conans/client/proxy.py]: Redundant?

File: conans/client/installer.py [hunk 3]
```diff
@@ -228,18 +232,25 @@ class ConanInstaller(object):
...
         self._build_mode = build_mode
         self._built_packages = set()  # To avoid re-building twice the same package reference
+        self._install_info = None
+        self._recorder = recorder
 
     def install(self, deps_graph, profile_build_requires, keep_build=False):
         """ given a DepsGraph object, build necessary nodes or retrieve them
         """
         t1 = time.time()
+        self._install_info = {"status": None,
```
[the remark this answers is not in the document]
Comment (lasote) [reply conans/client/installer.py]: nothing, previous approach not removed... programming at night is not good.

File: conans/client/proxy.py (hunk 2 above)
Comment (lasote) [reply conans/client/proxy.py]: yes

File: conans/client/proxy.py (hunk 1 above)
Comment (lasote) [reply conans/client/proxy.py]: Yes, I will pass always the recorder from the copy and uploader managers.

File: conans/client/action_recorder.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    def __init__(self):
+        self._inst_recipes_actions = OrderedDict()
+        self._inst_packages_actions = OrderedDict()
+
+    # ###### INSTALL METHODS ############
+
+    # RECIPE METHODS
+    def recipe_fetched_from_cache(self, reference):
+        if reference in self._inst_recipes_actions:
+            if self._inst_recipes_actions[reference].type == INSTALL_DOWNLOADED:
```
[the remark this answers is not in the document]
Comment (lasote) [reply conans/client/action_recorder.py]: It is but simplifies a lot this class. I'll try to improve it.

File: conans/client/proxy.py [hunk 5]
```diff
@@ -213,23 +218,35 @@ def _retrieve_from_remote(remote):
...
-                                                                             ref_remote.name))
+                msg = "%s was not found in remote '%s'" % (str(conan_reference), ref_remote.name)
+                self._recorder.recipe_install_error(conan_reference, INSTALL_ERROR_MISSING,
+                                                    msg, ref_remote.url)
+                raise NotFoundException(msg)
+            except RequestException as exc:
+                self._recorder.recipe_install_error(conan_reference, INSTALL_ERROR_NETWORK,
+                                                    str(exc), ref_remote.url)
+                raise exc
+            except Exception as exc:
```
Comment (memsharded) [inline conans/client/proxy.py]: This still redundant?

Comment (lasote) [reply conans/client/proxy.py]: uou. cleaning

**Pull request merged**